src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1523,7 +1563,13 @@ void ImportInitBlk()
 
         void ImportRethrow()
         {
-            // TODO:
+            if (_currentBasicBlock.HandlerIndex.HasValue &&
+             _exceptionRegions[_currentBasicBlock.HandlerIndex.Value].ILRegion.Kind == ILExceptionRegionKind.Catch)
```

Comment:
It would be nice to try to match the logic that is in the RyuJIT built-in verifier where possible: https://github.com/dotnet/coreclr/blob/master/src/jit/importer.cpp#L14490

Or in the original PEVerify if you prefer - https://github.com/lewischeng-ms/sscli/blob/master/clr/src/jit64/newverify.cpp#L3456 - but I believe that one it is harder to follow.